src/RequestOptions.php

```diff
@@ -151,6 +156,14 @@
     const MULTIPART = 'multipart';
 
     /**
+     * on_redirect: (callable) PHP callable that is invoked when a redirect
+     * is encountered. The callable is invoked with the request, the redirect
+     * response that was received, and the effective URI. Any return value
+     * from the on_redirect function is ignored.
+     */
+    const ON_REDIRECT = 'on_redirect';
```

Comment:
I'm -1 on adding `on_redirect` and `retries` to RequestOptions. These options are basically children of another option, and have no effect on sending a request unless you enable their parent options (e.g., `allows_redirects`) or their associated middleware (`RetryMiddleware`).